assets/helpers/fetch.js

```diff
@@ -1,8 +1,39 @@
 // @flow
 
+import { type Csrf as CsrfState } from 'helpers/csrf/csrfReducer';
+
+
+type Credentials = 'omit' | 'same-origin' | 'include';
+
+
 /** Sends a request to an API and converts the response into a JSON object */
 function fetchJson(endpoint: string, settings: Object): Promise<Object> {
   return fetch(endpoint, settings).then(resp => resp.json());
 }
 
-export { fetchJson };
+/** Builds a `RequestInit` object for use with GET requests using the Fetch API */
+function getRequestOptions(
+  credentials: Credentials,
+  csrf: CsrfState | null,
+): Object {
+  const headers = csrf !== null
```

Comment:
It's a shame JS does not provide a combinator to merge two or more records, but if useful we could write one, with the following laws:
```
right-unit: x ⩓ {} = x
left-unit: {} ⩓ x = x
associativity: (x ⩓ y) ⩓ z = x ⩓ (y ⩓ z)
```
let's call it `merge`: 
```js
merge(
  { 'Content-Type': 'application/json' }, 
  csrf !== null && csrf.token ? { 'Csrf-Token': csrf.token } : {}
)
```

Reply:
The potential implementations of `merge` are so simple that I wonder if it's worth defining the function?

```javascript
Object.assign(
  {},
  { 'Content-Type': 'application/json' },
  csrf !== null && csrf.token ? { 'Csrf-Token': csrf.token } : {}
)
```
or
```javascript
{
  ...{ 'Content-Type': 'application/json' },
  ...(csrf !== null && csrf.token ? { 'Csrf-Token': csrf.token } : {})
}
```
Though I suppose wrapping it in its own function makes it easier to compose